### PR TITLE
Update authors list

### PR DESCRIPTION
**Pull request by rolandlo**
Updated the list of authors obtained in raw form via 
```
 git shortlog -se   | perl -spe 's/^\s+\d+\s+//' > AUTHORS_raw
```
I have removed every member of the developer team (listed on the top) from the list of contributors (listed below) and removed duplicates and bots. 

Here is the diff between AUTHORS_raw and AUTHORS:
[diff.txt](https://github.com/xournalpp/xournalpp/files/10315780/diff.txt)

### CODE REVIEW

**Comment by Technius** [comment]
I would also update the dates on the maintenance team, and then update the About dialog text to be consistent with the list of maintainers.

**Review by Technius — APPROVED** [review]
We'll probably need to update the author list again when we prepare the 1.2.0 release...

**Comment by rolandlo** [comment]
I will squash the commits and rebase on master after merging #4970 (solving conflicts if necessary) in 24 hours if no objections are raised.

**Pull request merged**
